contracts/Auction.sol

```diff
@@ -166,9 +150,19 @@ contract Auction is IAuction {
     ///      one is to buy entire auction with the amount desired for this auction.
     ///      The other way is to buy a portion of an auction. In this case an
     ///      auction depleting rate is increased.
-    /// @param amount the amount the taker is paying, denominated in tokenAccepted
-    function takeOffer(uint256 amount) public override {
-        // TODO frontrunning mitigation
+    ///      WARNING: When calling this function directly, it might happen that
+    ///      the expected tokens to seize from the coverage pool is different
+    ///      from the actual one. There are a couple of reasons for that such as
+    ///      other bids taking this offer, claims or withdrawals on an Asset Pool
+    ///      that are executed in the same block. The recommended way for taking
+    ///      an offer is through 'AuctionBidder' contract with 'takeOfferWithMin'
+    ///      function, where a caller can specify the minimal value to receive
+    ///      from the coverage pool in exchange for its amount of tokenAccepted.
+    /// @param amount the amount the taker is paying, denominated in tokenAccepted.
+    ///               In the scenario when amount exceeds the outstanding tokens
+    ///               for the aution to complete, only the amount outstanding will
+    ///               be taken from a caller.
+    function takeOffer(uint256 amount) external override {
         require(amount > 0, "Can't pay 0 tokens");
```

Comment:
Linter complains about incorrect function order. Same in `CoveragePool.sol`.

Reply:
ff57359